tests/test_engine.py

```diff
@@ -181,6 +182,51 @@ def test_env_eval(self):
             if "BZT_ENV_TEST_UNSET" in os.environ:
                 os.environ.pop("BZT_ENV_TEST_UNSET")
 
+    def test_nested_env_eval(self):
+        try:
+            self.obj.config.merge({
+                "settings": {
+                    "env": {
+                        "FOO": "${BAR}/aaa/bbb",
+                        "FOOBAR": "eee",
+                        "BAR": "${BAZ}/ccc",
+                        "BAZ": "${FOOBAR}/ddd",
+                        "ART": "${FOO}===${TAURUS_ARTIFACTS_DIR}",
```

Comment:
There are still some bugs in algo.
For example here is the variables list which is not expanding as expected:

`{
  "J": "${K}/j",
  "K": "${L}/k",
  "L": "${P}/l",
  "P": "${O}/p",
  "O": "${I}/o",
  "I": "${U}/i",
  "U": "${Y}/u",
  "Y": "${T}/y",
  "T": "${R}/t",
  "R": "${E}/r",
  "E": "${W}/e",
  "W": "${Q}/w",
  "Q": "/q"
}`

Probably we can leave expanding logic as it was before and update logic for 'artifacts_dir' only?